app/controllers/submissions_controller.rb

```diff
@@ -321,6 +321,12 @@ def view
 
       @displayFilename = @submission.filename
     end
+
+    mm = MimeMagic.by_magic(file)
+    if not mm.text? and not mm.subtype == "pdf"
```

Comment:
I have pulled and tested this locally, and the issue was that it works fine with normal code file, but then for a completely normal Readme.txt file, MimeMagic returns nil.

I did some digging into the issue, and found out that was that MimeMagic's by magic function... is not so magical afterall. It basically does pattern matching based on https://github.com/minad/mimemagic/blob/master/lib/mimemagic/tables.rb

It did not identify the a file as a plain text because it does not match the any of the patterns they used for text

My suggestion is that we should use MimeMagic as a blacklist instead of a whitelist (to disallow "application" types) instead.

Reply:
Perhaps identify using regex, since MimeMagic only provides .text? but not for other types
`if mm.type =~ /application/`